src/Numerics/ODESolvers/DifferentialEquations.jl

```diff
@@ -0,0 +1,115 @@
+import DiffEqBase
+export DiffEqJLSolver, DiffEqJLIMEXSolver
+
+abstract type AbstractDiffEqJLSolver <: AbstractODESolver end
+
+"""
+    DiffEqJLSolver(f, RKA, RKB, RKC, Q; dt, t0 = 0)
+
+This is a time stepping object for explicitly time stepping the differential
+equation given by the right-hand-side function `f` with the state `Q`, i.e.,
+
+```math
+  \\dot{Q} = f(Q, t)
+```
+
+via a DifferentialEquations.jl DEAlgorithm, which includes support
+for OrdinaryDiffEq.jl, Sundials.jl, and more.
+"""
+mutable struct DiffEqJLSolver{I} <: AbstractDiffEqJLSolver
+    integ::I
+
+    function DiffEqJLSolver(rhs!,alg,Q,args...;
+                            t0 = 0, p = nothing,
+                            kwargs...)
+        prob = DiffEqBase.ODEProblem(
+                          (du,u,p,t)->rhs!(du,u,p,t; increment=false),
+                           Q, (float(t0),typemax(typeof(float(t0)))),
+                           p)
+        integ = DiffEqBase.init(prob,alg,args...;kwargs...)
+        new{typeof(integ)}(integ)
+    end
+end
+
+"""
+    DiffEqJLSolver(f, RKA, RKB, RKC, Q; dt, t0 = 0)
+
+This is a time stepping object for explicitly time stepping the differential
+equation given by the right-hand-side function `f` with the state `Q`, i.e.,
+
+```math
+  \\dot{Q} = f_I(Q, t) + f_E(Q, t)
+```
+
+via a DifferentialEquations.jl DEAlgorithm, which includes support
+for OrdinaryDiffEq.jl, Sundials.jl, and more.
+"""
+mutable struct DiffEqJLIMEXSolver{I} <: AbstractDiffEqJLSolver
+    integ::I
+
+    function DiffEqJLIMEXSolver(rhs!,rhs_implicit!,alg,Q,args...;

```

Comment:
it probably would make sense to switch all our solvers so that the implicit part is first. this is probably going to be more important once we want multirate.